tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -0,0 +1,303 @@
+        self.assertEqual(process_field("foo", [], "usage"), "foo")
+
+
+class SubscriptionInterfaceTestCase(unittest.TestCase):
+    """Test DBus interface for the subscription module."""
+
+    def setUp(self):
+        """Set up the subscription module."""
+        self.subscription_module = SubscriptionService()
+        self.subscription_interface = SubscriptionInterface(self.subscription_module)
+
+        # Connect to the properties changed signal.
+        self.callback = PropertiesChangedCallback()
+        self.subscription_interface.PropertiesChanged.connect(self.callback)
+
+    def _test_dbus_property(self, *args, **kwargs):
+        check_dbus_property(
+            self,
+            SUBSCRIPTION,
+            self.subscription_interface,
+            *args, **kwargs
+        )
+
+    def kickstart_properties_test(self):
+        """Test kickstart properties."""
+        self.assertEqual(self.subscription_interface.KickstartCommands, ["syspurpose", "rhsm"])
+        self.assertEqual(self.subscription_interface.KickstartSections, [])
+        self.assertEqual(self.subscription_interface.KickstartAddons, [])
+        self.callback.assert_not_called()
+
+    def system_purpose_data_test(self):
+        """Test the SystemPurposeData DBus structure."""
+
+        # create the SystemPurposeData structure
+        system_purpose_data = SystemPurposeData()
+        system_purpose_data.role = "foo"
+        system_purpose_data.sla = "bar"
+        system_purpose_data.usage = "baz"
+        system_purpose_data.addons = ["a", "b", "c"]
+
+        # create its expected representation
+        expected_dict = {
+            "role": get_variant(Str, "foo"),
+            "sla": get_variant(Str, "bar"),
+            "usage": get_variant(Str, "baz"),
+            "addons": get_variant(List[Str], ["a", "b", "c"])
+        }
+
+        # compare the two
+        self.assertEqual(SystemPurposeData.to_structure(system_purpose_data), expected_dict)
+
+    def set_system_purpose_test(self):
+        """Test if setting system purpose data from DBUS works correctly."""
+        system_purpose_data = {
+            "role": get_variant(Str, "foo"),
+            "sla": get_variant(Str, "bar"),
+            "usage": get_variant(Str, "baz"),
+            "addons": get_variant(List[Str], ["a", "b", "c"])
+        }
+
+        self._test_dbus_property(
+          "SystemPurposeData",
+          system_purpose_data
+        )
+
+        output_structure = self.subscription_interface.SystemPurposeData
+        output_system_purpose_data = SystemPurposeData.from_structure(output_structure)
+
+        self.assertEqual(output_system_purpose_data.role, "foo")
+        self.assertEqual(output_system_purpose_data.sla, "bar")
+        self.assertEqual(output_system_purpose_data.usage, "baz")
+        self.assertEqual(output_system_purpose_data.addons, ["a", "b", "c"])
+
+    def _test_kickstart(self, ks_in, ks_out):
```

Comment:
For private test method it would be nice to name them `check` instead of test. To make it clear that these are not test units.

Reply:
As with the license in `__main__.py` file - this is how we currently do it in most if not all tests. If we want to change it we should do it in a separate PR for all tests. :)